### PR TITLE
Rich text: remove space handling (for BUTTON and SUMMARY)

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Seems to do more harm than good. We currently also don't have any blocks using rich text with BUTTON or SUMMARY as a tag name. Also work on the details block seems to be abandoned (#23940). There might be plugins using these tag names though...

Alternatively we could just return early for all other tag names.

### CODE REVIEW

File: packages/rich-text/src/component/use-space.js [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-import { SPACE } from '@wordpress/keycodes';
-
-/**
- * For some elements like BUTTON and SUMMARY, the space key doesn't insert a
- * space character in some browsers even though the element is editable. We have
- * to manually insert a space and prevent default behaviour.
- *
- * DO NOT limit this behaviour to specific tag names! It would mean that this
- * behaviour is not widely tested. If there's ever any problems, we should find
- * a different solution entirely or remove it entirely.
```
Comment (ellatrix) [inline packages/rich-text/src/component/use-space.js:14]: It was a nice try, but this solution clearly failed. Cc @dmsnell

**Comment by dmsnell** [comment]
> Also work on the details block seems to be abandoned (https://github.com/WordPress/gutenberg/pull/23940).

On my list but in my version I'm not using `<summary>` for the many issues it carries. I don't even think it's the right tag in a general sense for the editing process so I don't consider avoiding it a compromise.

> We currently also don't have any blocks using rich text with BUTTON or SUMMARY as a tag name.

Thanks for checking. I guess it could be possible that some integrations re-surface the issue with space not entering a space.

For reference, I double-checked behaviors:

|| Safari | Firefox | Chromium |
|---|---|---|---|
| `<button>` | Adds space | Adds space | Clicks button, no space |
| `<summary>` | Toggles details view, no space | Toggles details view, adds space | Toggles details view, no space |

This was tested with a basic HTML file.

```html
<div contenteditable>This is content.</div>

<button contenteditable>This is content.</button>


<details>
  <summary contenteditable>This can be edited</summary>
  But this is static.
</details>
```

I think we can probably handle this at those tags if and when it arises maybe. Both the button and summary have problems in edit mode where it's less important anyway that we use those specific tags (my argument is based on the claim that semantics while editing are different than when viewing).

It could be best if we warn on `RichText` not to use those tags, to prefer something else instead.

**Review by dmsnell — APPROVED** [review]
While this is still a draft I think it's a good call to gut it. I wish we had a good way to leave a tombstone in the code to say, "hey, we tried this and it failed" because I think the `git` history won't be easy to find. I don't have a good idea on how to do this so I'm not suggesting we block merging this until we find one, but if you have an idea it would be worth tossing it in I think.

**Comment by ellatrix** [comment]
> For reference, I double-checked behaviors:

Did you check with `event.preventDefault()`? The problem is that we want to prevent the toggling behaviour, but not the space inserting behaviour. Unfortunately there's no way to do that.

I guess another solution for `summary` could be to simply undo the toggling behaviour or somehow force keeping it open.

I do think it's good to use the tags if possible in the future, because it's easier to style by themes. But anyway, it's not important right now.

**Comment by dmsnell** [comment]
> Did you check with event.preventDefault()?

No I didn't, because…

> The problem is that we want to prevent the toggling behaviour, but not the space inserting behaviour. Unfortunately there's no way to do that.

…anyway

> I guess another solution for summary could be to simply undo the toggling behaviour or somehow force keeping it open.
> I do think it's good to use the tags if possible in the future, because it's easier to style by themes. But anyway, it's not important right now.

My point about avoiding these tags is _during the editing operation_ and I think all the styling concerns are diminished there. We know in the editor we have JavaScript and React and if we really want a collapsable editing container without the baggage of `<details>` we can create that, but we don't need `RichText` to support those environments. As soon as `isSelected` for a block is `false` (or as soon as someone `onBlur`s the part they're editing) inside of `edit()` we can render a `<button>` or `<details>` if we must (though the `<summary>` still poses problems with keyboard navigation in the editor).

We don't have to insist on the same implementation (`<details>`) through all levels of the stack to preserve the visual consistency with the rendered page.

**Pull request merged**